helix-term/src/commands.rs

```diff
@@ -2659,9 +2661,85 @@ fn last_picker(cx: &mut Context) {
 }
 
 // I inserts at the first nonwhitespace character of each line with a selection
+// If the line is empty, automatically indent
 fn insert_at_line_start(cx: &mut Context) {
-    goto_first_nonwhitespace(cx);
     enter_insert_mode(cx);
+
+    let (view, doc) = current!(cx.editor);
+
+    let text = doc.text().slice(..);
+    let contents = doc.text();
+    let selection = doc.selection(view.id);
+
+    let language_config = doc.language_config();
+    let syntax = doc.syntax();
+    let tab_width = doc.tab_width();
+
+    let mut ranges = SmallVec::with_capacity(selection.len());
+    let mut offs = 0;
+
+    let mut transaction = Transaction::change_by_selection(contents, selection, |range| {
+        let cursor = range.from();
+        let cursor_line = text.char_to_line(cursor);
+
+        let doc_end = contents.len_chars();
+        if range.from() == doc_end {
+            ranges.push(Range::point(doc_end + offs));
+            return (doc_end, doc_end, None);
+        }
```

Comment:
```suggestion
```

I don't think this is correct, the EOF position is not always it's own line. If the file doesn't end with a newline terminator the EOF is part of a normal line and `I` should work in that case too.
This condition is not even required because all functions in helix handle both the one past the end char index and one past the end line just fine (And I think we should also auto indent in that case because everything else would be inconsistent)

Reply:
Yup, this is not required anymore now.